rerun_py/rerun_sdk/rerun/log/rects.py

```diff
@@ -182,71 +141,60 @@ def log_rects(
         See also: [`rerun.init`][], [`rerun.set_global_data_recording`][].
 
     """
-    from rerun.experimental import cmp as rrc
+    from rerun.experimental import Boxes2D, log
 
-    recording = RecordingStream.to_native(recording)
+    if rects is None:
+        raise ValueError("`rects` argument must be set")
 
-    # Treat None the same as []
     if np.any(rects):  # type: ignore[arg-type]
         rects = np.asarray(rects, dtype="float32")
+        if rects.ndim == 1:
+            rects = np.expand_dims(rects, axis=0)
     else:
         rects = np.zeros((0, 4), dtype="float32")
     assert type(rects) is np.ndarray
 
-    colors = _normalize_colors(colors)
-    class_ids = _normalize_ids(class_ids)
-    labels = _normalize_labels(labels)
+    if rect_format == RectFormat.XYWH:
+        half_extents = rects[:, 2:4] / 2
+        centers = rects[:, 0:2] + half_extents
+    elif rect_format == RectFormat.YXHW:
+        half_extents = rects[:, 4:2] / 2
+        centers = rects[:, 2:0] + half_extents
+    elif rect_format == RectFormat.XYXY:
+        left_top = rects[:, 0:2]
+        right_bottom = rects[:, 2:4]
+        centers = (left_top - right_bottom) / 2
+        half_extents = right_bottom - centers
```

Comment:
there is a bug here in the center calculation!

But I also think we should be using `min` and `max`. `left_top` etc unnecessarily implies a specific coordinate system

```suggestion
        min = rects[:, 0:2]
        max = rects[:, 2:4]
        centers = (min + max) / 2.0
        half_extents = max - centers
```

Reply:
great catch, thank you!